xchange-stream-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioStreamingRawService.java

```diff
@@ -183,6 +184,13 @@ protected void handleMessage(JsonNode message) {
             }
             break;
           case PING:
+            // Sometimes the timestamp sent by Cex IO is up to a second ahead of current time. Then our pong sends a
+            // timestamp before Cex IO's own time.
+            // When this happens, Cex IO seems to ignore the pong and closes the connection 20-30 seconds later
+            long time = message.get("time").longValue();
+            while (Instant.now().isBefore(Instant.ofEpochMilli(time))) {
+              Thread.yield();
```

Comment:
This is still basically busy-waiting an event loop thread. If we really need to wait it would be better simply to [schedule the response](https://netty.io/4.0/api/io/netty/util/concurrent/EventExecutorGroup.html#schedule-java.lang.Runnable-long-java.util.concurrent.TimeUnit-), but I'd like to understand why we can't just sync the clock that's generating the problematic timestamp.